oaknational/Oak-Web-Application
# ENG-391 feat(canonical lesson): canonical lesson page

## Description

- adds new curriculum api queries and pages (lesson-overview-canonical, lesson-downloads-canonical)
  - they use the same MVs as lesson-overview and lesson-downloads
  - it groups instances of the same lesson together, adding a list of pathways in which the lesson is found
- Adds components (with storybook stories, [should match Figma](https://www.figma.com/file/3i0y2PHT6pECaGgvLhoN8z/Lesson-page-July-2023?node-id=2111%3A54147&mode=dev)) for:
  - `TagFunctional`
  - `LessonAppearsIn`
  - `LessonPathwayCard`
- organises the files a little into a `@/components/Lesson` directory
- amended some Avo events so that some of pathway fields are optional, and those changes are reflected in this PR 




## How to test
- go to https://deploy-preview-1891--oak-web-application.netlify.thenational.academy/teachers/lessons/masculinity-as-macbeth-s-hamartia
- you should see links to the lesson page within three units at the bottom
- got to https://deploy-preview-1891--oak-web-application.netlify.thenational.academy/teachers/lessons/transverse-waves
- you should see units separated by subject like:
<img width="1440" alt="image" src="https://github.com/oaknational/Oak-Web-Application/assets/12934669/e5dd01e1-d80b-42fe-a49d-622eafb0a256">
